src/python/pants/core/goals/fmt.py

```diff
@@ -130,23 +143,80 @@ def files(self) -> tuple[str, ...]:
             return self.elements
 
     @classmethod
-    def _get_registration_rules(cls) -> Iterable[UnionRule]:
-        yield from super()._get_registration_rules()
+    def _get_registration_rules(cls, *, partitioner_type: PartitionerType) -> Iterable[UnionRule]:
+        yield from super()._get_registration_rules(partitioner_type=partitioner_type)
         yield UnionRule(FmtRequest, cls)
         yield UnionRule(FmtRequest.SubPartition, cls.SubPartition)
 
 
 class FmtTargetsRequest(FmtRequest, LintTargetsRequest):
+    @memoized_classproperty
+    def _default_single_partition_partitioner_rules(cls) -> Iterable:
+        # NB: This only works if the FieldSet has a single `SourcesField` field. We check here for
+        # a better user experience.
```

Comment:
Pants generally fails already if you have multiple sources fields on a target. But doesn't hurt to be defensive here

Reply:
It doesn't work. It would probably be good to explicitly error, even if that meant special-cased code inside the `Target.py` constructor..

Reply:
https://github.com/pantsbuild/pants/issues/17132